File: FindWeather/DPLab08-Refactored/src/main/java/Model/Weather.java
package Model;

public class Weather {
    public Location location;
    public double temperature;
    public String weather_condition;
    public String WeatherAPI_name;

    public Weather(double temperature, String weather_condition, Location location, String WeatherAPI_name) {
        this.temperature = temperature;
        this.weather_condition = weather_condition;
        this.location = location;
        this.WeatherAPI_name = WeatherAPI_name;
    }

    public void DisplayWeather() {
        System.out.println("Weather in " + location.getCity() + " from " + WeatherAPI_name + ": "
                + temperature + "°C, " + weather_condition);
    }
}

File: FindWeather/Lab-8_FindWeather/Weather.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_8_FindWeather
{
    public class Weather
    { 
        public Location location;
        public double Temperature { get; set; } 
        public string Conditions { get; set; }
        public Location Location { get; set; }
        public string Source { get; set; }

        public Weather(double Temperature, string Conditions, Location Location, string Source)
        {
            this.Temperature = Temperature;
            this.Conditions = Conditions;
            this.Location = Location;
            this.Source = Source;
        }

        public Weather(Location location)
        {
            this.location = location;
        }

        public void DisplayWeather()
        {
            Console.WriteLine($"Weather in {Location.GetCity()} from {Source}: {Temperature}°C, {Conditions}\n");
        }
    }
}
